packages/ai-jsx/src/core/core.ts

```diff
@@ -444,9 +447,17 @@ function createRenderContextInternal(render: StreamRenderer, userContext: Record
 
         [Symbol.asyncIterator]: () => {
           if (hasReturnedGenerator) {
-            throw new Error("The RenderResult's generator was already returned and cannot be returned again.");
+            throw new AIJSXError(
+              "The RenderResult's generator was already returned and cannot be returned again.",
+              1009,
+              'internal'
```

Comment:
This error indicates a bug in a component (namely, there are restrictions on how you can use a `RenderResult`, and these are thrown if those restrictions are violated).

Reply:
unknown